File: src/main/java/ru/openbank/meetup/demo/java/slide3/SerializableOtpDto.java
package ru.openbank.meetup.demo.java.slide3;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.validation.constraints.Pattern;
import java.io.Serializable;

@Data
@NoArgsConstructor
@AllArgsConstructor
@ApiModel("Otp")
@JsonIgnoreProperties(ignoreUnknown = true)
public class SerializableOtpDto implements Serializable {

    private static final long serialVersionUID = -1069041605866219052L;

    @ApiModelProperty(example = "1234")
    @Pattern(regexp = "[0-9]{4}")
    private String otp;

    @ApiModelProperty(example = "71112223344")
    @Pattern(regexp = "7[0-9]{10}")
    private String phone;

}

File: src/main/kotlin/ru/openbank/meetup/demo/kotlin/slide3/SerializableOtpDto.kt
package ru.openbank.meetup.demo.kotlin.slide3

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import io.swagger.annotations.ApiModel
import io.swagger.annotations.ApiModelProperty
import java.io.Serializable
import javax.validation.constraints.Pattern

@ApiModel("Otp")
@JsonIgnoreProperties(ignoreUnknown = true)
data class SerializableOtpDto(

    @field:ApiModelProperty(example = "1234")
    @field:Pattern(regexp = "[0-9]{4}")
    var otp: String? = null,

    @field:ApiModelProperty(example = "71112223344")
    @field:Pattern(regexp = "7[0-9]{10}")
    var phone: String? = null
) : Serializable {
    companion object {
        private const val serialVersionUID = -1069041605866219052L
    }
}

